### ISSUE TITLE
S2252 FP: Rule raises issue in case of non integer value in for loop.

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#5428]
### Description

S2252 raises issue in case of non integer value in for loop. Issue was raised  by community, see https://community.sonarsource.com/t/this-loop-will-never-execute-will-execute-with-negative-and-decrementing/58753

### Repro steps

```
for (var n = 0.0; n > -0.1; n -= 0.005)
{
}
```

### Expected behavior

No issue is raised.

### Actual behavior

An issue is raised.

### Known workarounds

None

### Related information

* C#/VB.NET Plugins version: 8.36.0.43782
